src/overlays/actors/ovl_En_Am/z_en_am.c

```diff
@@ -93,63 +112,881 @@ static ColliderQuadInit D_809AFFD8 = {
+    /* 01 */ AM_DMGEFF_NUT,
+    /* 06 */ AM_DMGEFF_STUN = 6, // doesnt include deku nuts
+    /* 13 */ AM_DMGEFF_ICE = 13,
+    /* 14 */ AM_DMGEFF_MAGIC_FIRE_LIGHT,
+    /* 15 */ AM_DMGEFF_KILL // any damage source that can kill the armos (and isnt a special case)
+} ArmosDamageEffect;
+
+static DamageTable sDamageTable = {
+    /* Deku nut      */ DMG_ENTRY(0, AM_DMGEFF_NUT),
+    /* Deku stick    */ DMG_ENTRY(2, AM_DMGEFF_NONE),
+    /* Slingshot     */ DMG_ENTRY(1, AM_DMGEFF_NONE),
+    /* Explosive     */ DMG_ENTRY(2, AM_DMGEFF_KILL),
+    /* Boomerang     */ DMG_ENTRY(0, AM_DMGEFF_STUN),
+    /* Normal arrow  */ DMG_ENTRY(2, AM_DMGEFF_KILL),
+    /* Hammer swing  */ DMG_ENTRY(2, AM_DMGEFF_KILL),
+    /* Hookshot      */ DMG_ENTRY(0, AM_DMGEFF_STUN),
+    /* Kokiri sword  */ DMG_ENTRY(1, AM_DMGEFF_NONE),
+    /* Master sword  */ DMG_ENTRY(2, AM_DMGEFF_KILL),
+    /* Giant's Knife */ DMG_ENTRY(4, AM_DMGEFF_KILL),
+    /* Fire arrow    */ DMG_ENTRY(2, AM_DMGEFF_KILL),
+    /* Ice arrow     */ DMG_ENTRY(4, AM_DMGEFF_ICE),
+    /* Light arrow   */ DMG_ENTRY(2, AM_DMGEFF_KILL),
+    /* Unk arrow 1   */ DMG_ENTRY(2, AM_DMGEFF_NONE),
+    /* Unk arrow 2   */ DMG_ENTRY(2, AM_DMGEFF_NONE),
+    /* Unk arrow 3   */ DMG_ENTRY(2, AM_DMGEFF_NONE),
+    /* Fire magic    */ DMG_ENTRY(0, AM_DMGEFF_MAGIC_FIRE_LIGHT),
+    /* Ice magic     */ DMG_ENTRY(3, AM_DMGEFF_ICE),
+    /* Light magic   */ DMG_ENTRY(0, AM_DMGEFF_MAGIC_FIRE_LIGHT),
+    /* Shield        */ DMG_ENTRY(0, AM_DMGEFF_NONE),
+    /* Mirror Ray    */ DMG_ENTRY(0, AM_DMGEFF_NONE),
+    /* Kokiri spin   */ DMG_ENTRY(1, AM_DMGEFF_NONE),
+    /* Giant spin    */ DMG_ENTRY(4, AM_DMGEFF_KILL),
+    /* Master spin   */ DMG_ENTRY(2, AM_DMGEFF_KILL),
+    /* Kokiri jump   */ DMG_ENTRY(2, AM_DMGEFF_NONE),
+    /* Giant jump    */ DMG_ENTRY(8, AM_DMGEFF_KILL),
+    /* Master jump   */ DMG_ENTRY(4, AM_DMGEFF_KILL),
+    /* Unknown 1     */ DMG_ENTRY(0, AM_DMGEFF_NONE),
+    /* Unblockable   */ DMG_ENTRY(0, AM_DMGEFF_NONE),
+    /* Hammer jump   */ DMG_ENTRY(4, AM_DMGEFF_KILL),
+    /* Unknown 2     */ DMG_ENTRY(0, AM_DMGEFF_NONE),
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Am/func_809AEFA4.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_S8(naviEnemyId, 19, ICHAIN_CONTINUE),
+    ICHAIN_F32_DIV1000(gravity, -4000, ICHAIN_CONTINUE),
+    ICHAIN_F32(targetArrowOffset, 5300, ICHAIN_STOP),
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Am/func_809AF0DC.s")
+void EnAm_SetupAction(EnAm* this, EnAmActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+/**
+ * Checks bgCheckFlags in the direction of current yaw at the specified distance.
+ *
+ * Returns true if the armos would land on the ground in the resulting position.
+ *
+ * If it won't land on the ground, it will return true anyway if the floor the armos will be on
+ * is 20 units higher than the home position. This is a convoluted way of making sure
```

Comment:
20 units *lower*, right?

Reply:
no, higher 
`this->dyna.actor.floorHeight >= (this->dyna.actor.home.pos.y - 20.0f)`, it returns true

Reply:
If the armos is currently at height 100, it will return true if the floor is at least at height 80.

Reply:
still not really following

if the floor height it was going to land on is 20 or more units lower than its home position it would return false as to prevent it from going down steep slopes

Reply:
Right, but the phrase `if the floor the armos will be on is 20 units higher than the home position.` implies the floor itself is 20 units higher than the home position (i.e. at height 120 in my example), instead of being no more than 20 units lower.

Reply:
ooo yea i see where i was tripping myself up lmao, alright thanks